packages/medusa/src/api/routes/store/carts/create-payment-sessions.ts

```diff
@@ -26,17 +27,94 @@ export default async (req, res) => {
   const { id } = req.params
 
   const cartService: CartService = req.scope.resolve("cartService")
-
+  const idempotencyKeyService: IdempotencyKeyService = req.scope.resolve(
+    "idempotencyKeyService"
+  )
   const manager: EntityManager = req.scope.resolve("manager")
-  await manager.transaction(async (transactionManager) => {
-    return await cartService.withTransaction(transactionManager).setPaymentSessions(id)
-  })
 
-  const cart = await cartService.retrieve(id, {
-    select: defaultStoreCartFields,
-    relations: defaultStoreCartRelations,
-  })
+  const headerKey = req.get("Idempotency-Key") || ""
+
+  let idempotencyKey
+  try {
+    await manager.transaction(async (transactionManager) => {
+      idempotencyKey = await idempotencyKeyService.withTransaction(transactionManager).initializeRequest(
+        headerKey,
+        req.method,
+        req.params,
+        req.path
+      )
+    })
+  } catch (error) {
+    res.status(409).send("Failed to create idempotency key")
+    return
+  }
+
+  res.setHeader("Access-Control-Expose-Headers", "Idempotency-Key")
+  res.setHeader("Idempotency-Key", idempotencyKey.idempotency_key)
+
+  try {
+    let inProgress = true
+    let err: unknown = false
+
+    while (inProgress) {
+      switch (idempotencyKey.recovery_point) {
+        case "started": {
+          await manager.transaction(async (transactionManager) => {
+            const { key, error } = await idempotencyKeyService
+              .withTransaction(transactionManager)
+              .workStage(
+                idempotencyKey.idempotency_key,
+                async (manager) => {
+                  await cartService.withTransaction(manager).setPaymentSessions(id)
+
+                  const cart = await cartService.withTransaction(manager).retrieve(id, {
```

Comment:
question: Just want to be sure that parameter `manager` does not conflict with the declaration of const `manager` resolved from Awilix on line 33?

Reply:
It does not because it takes the one from the actual scope in that case.
```
const manager = "original manager"

const func = (manager) => {
  console.log(manager)
}

func("scoped_manager")
console.log(manager)
```

I did that just to check that what I thought was right.

wdyt if I still rename it just to be absolutely sure :)

Reply:
Thanks for clarifying!